CreeDictionary/CreeDictionary/hfstol.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+"""
+Temporary HFSTOL.
+"""
+
+import shutil
+from contextlib import contextmanager
+from subprocess import DEVNULL, check_output
+from tempfile import TemporaryFile
+from typing import IO, Generator, Iterable, NamedTuple
+
+from utils.shared_res_dir import shared_res_dir as res
+
+# Ensure we can load everything!
+_analyzer_path = res / "fst" / "crk-descriptive-analyzer.hfstol"
+assert _analyzer_path.exists()
+_hfstol_bin = shutil.which("hfst-optimized-lookup")
+if _hfstol_bin is None:
+    raise ImportError("Cannot find hfst-optimized-lookup! Is it installed?")
+else:
+    _hfstol = _hfstol_bin
+
+
+class Analysis(NamedTuple):
+    """
+    Analysis of a wordform.
+    """
+
+    raw_prefixes: str
+    lemma: str
+    raw_suffixes: str
+
+
+@contextmanager
+def write_file(text: str) -> Generator[IO[str], None, None]:
+    with TemporaryFile("w+", encoding="UTF-8") as tmp:
+        tmp.write(text)
+        tmp.flush()
+        tmp.seek(0)
+        yield tmp
+
+
+def analyze(wordform: str) -> Generator[Analysis, None, None]:
```

Comment:
aside from descriptive analyzer, normative generator is also used in the [search code] (https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/blob/cf6057755f949501c5dc728fb2e1be55431f1c3e/CreeDictionary/API/models.py#L382-L403) to generate normative forms when the user gives analyzable words outside of the paradigm.

so we need a `generate` function. Do you think we need a separate file, a separate function in this same file or do you prefer other methods?

Reply:
I'll write a separate `generate()` function in this file!